src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -271,27 +283,71 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (contain_mutable_functions((Node *) targetEntry->expr))
+			if (commandType == CMD_UPDATE &&
+				contain_volatile_functions((Node *) targetEntry->expr))
 			{
-				hasNonConstTargetEntryExprs = true;
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("functions used in UPDATE queries on distributed "
```

Comment:
I added a comment for INSERT but not one for INSERT INTO SELECT. Where in INSERT INTO SELECT would you fit a mutable function?